plugins/prometheus_plugin/metrics.hpp

```diff
@@ -116,6 +135,35 @@ struct catalog_type {
    void update(const net_plugin::p2p_connections_metrics& metrics) {
       num_peers.Set(metrics.num_peers);
       num_clients.Set(metrics.num_clients);
+      for(size_t i = 0; i < metrics.stats.peers.size(); ++i) {
+         auto addr = boost::asio::ip::make_address_v6(metrics.stats.peers[i].address).to_string();
+         boost::replace_all(addr, ":", "_COLON_");
+         boost::replace_all(addr, ".", "_DOT_");
```

Comment:
This gets presented nicely in `net-util.py` because you convert them back. But doesn't this cause it to be displayed rather ugly in say something like Grafana?

Reply:
Yes, unavoidably.  The only characters allowed in a Prometheus label are alphanumerics and underscore.  Neither colon nor dots can be used in labels.  This limitation is part of the specification and is enforced by the library we're using for marshalling Prometheus data.

Reply:
Maybe the unique identifier should be connection id with one of the values being the address.